lib/blocs/change_password_bloc/change_password_bloc.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:news_app/services/change_password_services.dart';
+
+import '../../src/models/my_user.dart';
+
+part 'change_password_event.dart';
+part 'change_password_state.dart';
+
+class ChangePasswordBloc
+    extends Bloc<ChangePasswordEvent, ChangePasswordState> {
+  final FirebaseFirestore _instance = FirebaseFirestore.instance;
+  final FirebaseAuth _user = FirebaseAuth.instance;
+  final ChangePasswordServices _changePasswordServices =
+      ChangePasswordServices();
+  ChangePasswordBloc() : super(const ChangePasswordState.init()) {
+    on<ChangePassword>(_onChangePassword);
+  }
+  void _onChangePassword(
+      ChangePassword event, Emitter<ChangePasswordState> emit) async {
+    var data = await _instance
+        .collection("Users")
+        .where("uid", isEqualTo: _user.currentUser!.uid)
+        .get();
+    var usersData = MyUser.fromJson(data.docs[0].data());
```

Comment:
check lại logic xử lý đoạn này `data.docs[0]`

Reply:
em đã thêm vào is Empty + check exception